Train a baseline model from scratch on `dcm-zurich-lesions` datasets using nnUNet

valosekj opened the pull request:
This PR contains scripts for training baseline models from scratch on `dcm-zurich-lesions` datasets using nnUNetv2:
- region-based model: [#1](https://github.com/ivadomed/model-seg-dcm/issues/1)
- multi-channel model: [#5](https://github.com/ivadomed/model-seg-dcm/issues/5)

valosekj committed:
Add script to generate missing SC GT (using the SCIseg model)

valosekj committed:
Add convert_bids_to_nnUNetv2.py script.

valosekj committed:
rename 'train_subjects' --> 'train_images'

valosekj committed:
rename 'test_subjects' --> 'test_images'

valosekj committed:
change '--region-based' to default=True

valosekj committed:
update suffixes to be compatible with dcm-zurich-lesions datasets

valosekj committed:
make clear the script work with images not subjects

valosekj committed:
make clear the script work with images not subjects

valosekj committed:
Add print

valosekj committed:
remove empty line

valosekj committed:
update description

valosekj commented on dataset_conversion/utils.py:
```diff
@@ -0,0 +1,781 @@
+    # create a new label array with the same shape as the original labels
+    label_npy = np.zeros(lesion_label_npy.shape, dtype=np.int16)
+    # spinal cord
+    label_npy[seg_label_npy == 1] = 1
+    # lesion seg
+    label_npy[lesion_label_npy == 1] = 2
```
@naga-karthik I was thinking about using `sct_maths -add` to make sure that the lesion is part of the SC seg (as we did this for SCIseg). But now I am reading these two lines. Basically, line 45 sets the value `2` to the lesion regardless of whether the lesion is part of the SC or not. Meaning that `sct_maths -add` is unnecessary?

naga-karthik replied:
Yeah, you're right! Because this was already happening during nnunet conversion, I thought that you were doing `sct_maths -add` just as a sanity check to ensure lesion is inside the SC!

valosekj replied:
Okay, great, thanks for confirmation!

valosekj committed:
Add nnUNetv2 training scripts

valosekj committed:
Improve print

valosekj committed:
Add example usage

valosekj committed:
update permissions to 770

valosekj committed:
Fix input args for training/02_compute_anima_metrics.py

valosekj committed:
Make explicit the script converts data to REGION-BASED format

valosekj committed:
Fix typo

valosekj committed:
Fix example usage

valosekj committed:
update '--dataset-name' arg default value

valosekj committed:
remove obsolete condition

valosekj committed:
update 'create_region_based_label' docstring

valosekj committed:
Add script for MULTI-CHANNEL trianing and update utils.py

valosekj committed:
remove unnecessary code

valosekj committed:
refactor the code to make it compatible with both region-based and multi-channel training

valosekj committed:
Add the 'nnUNetTrainerDiceCELoss_noSmooth' trainer

valosekj committed:
Print number of training and validation images for each site

valosekj committed:
udpate help

valosekj committed:
include seed into dataset name

valosekj committed:
Add 'seed'

valosekj committed:
Remove 'seed' -- seems not to be compatible with 'nnunetv2/utilities/dataset_name_id_conversion.py':

"raise RuntimeError("More than one dataset name found for dataset id %d. Please correct that."

valosekj committed:
Revert "Remove 'seed' -- seems not to be compatible with 'nnunetv2/utilities/dataset_name_id_conversion.py':"

This reverts commit 73b9614fef5f2b49019397e72a8a14d7302ee6a3.

valosekj committed:
Update 'dataset_num'

naga-karthik commented on dataset_conversion/convert_bids_to_nnUNetv2_region-based.py:
```diff
@@ -0,0 +1,432 @@
+
+    # Needed for finding the files correctly. IMPORTANT! File endings must match between images and segmentations!
+    json_dict['file_ending'] = ".nii.gz"
+
```
Just to be consistent with latest findings about axes swaps -- maybe we should specify the IO reader ? 

```
json_dict['overwrite_image_reader_writer'] = "NibabelIO"
```

valosekj replied:
Good point! I've been thinking about that too! My only question is: will models with different IO readers (Nibabel/SimpleITK) be comparable?

valosekj committed:
fix typo

valosekj committed:
fix example usage

valosekj committed:
remove unused import

valosekj committed:
fix comment
